src/core/ddsc/tests/time.c

```diff
@@ -0,0 +1,17 @@
+#include <criterion/criterion.h>
+#include <criterion/logging.h>
+
+#include "ddsc/dds.h"
+
+Test(ddsc_time, request_time)
+{
+    dds_time_t now, then;
+    dds_duration_t pause = 1 * DDS_NSECS_IN_MSEC;
+
+    now = dds_time();
+    cr_assert_gt(now, 0);
+    /* Sleep for 1 millisecond, every platform should (hopefully) support that */
+    dds_sleepfor(pause);
+    then = dds_time();
+    cr_assert_gt(then, now);
```

Comment:
The test could be strengthed: if it is required that ``then > now``, then ``then >= now + pause`` should hold as well.

And what about platforms that can't sleep for a 1ms — do we guarantee that sleepfor will then sleep for longer, or does it merely yield the CPU? Sleeping a bit longer would avoid that trouble, so why not increase ``pause`` to something that really "should" be supported?